### DISCUSSION TITLE
working with existing database structure and it's relations (postgres)

### DISCUSSION THREAD

**Discussion by melistik** [Budibase/budibase#9407] [Help]
Hi,
i'm new to budibase and tested the tool locally with my aleady existing datamodel.
The plan was to build a little admin-ui based on my application's datastructure.

In general i would like to use the auto-ui generation with the knowledge about relations ;)

I established a connection easily to my postgres-db, entities appear easily - relations haven't been deteced although fk + indices are existing. When i try to remapped relations i need to create new columns. Is their any approach to inform budibase about aleady existing relations? I don't want to recreate my datamodel by added extra columns and remove the existing one. Furthermore the scanning and usage of existing models feels a bit useless without this feature.

Hope my usecase is understandable and their is a workaround for this issue?

When I download my model and extract the zip-file i propably could fake the relations within the json but for a bigger model this feels unhandlebale.

Thx in advance Marten

**Comment by melohagan** [comment]
Hey Marten,

I did create an issue related to this recently: https://github.com/Budibase/budibase/issues/9287

Currently when you [define relationships](https://docs.budibase.com/docs/sql-datasource#defining-relationships) the new columns are only created within Budibase to indicate the relationship. It doesn't affect your underlying SQL schema.

If you have a large number of relationships, I recommend that you use custom SQL queries instead to pull the specific data you need for your screen: https://docs.budibase.com/docs/custom-queries
